avl_template_new.py

```diff
@@ -1072,21 +1072,45 @@ def shuffle(self, array):
 	@returns: the absolute value of the difference between the height of the AVL trees joined
 	"""
 	def concat(self, lst):
-		absDiff = abs(self.root.height - lst.root.height)
+		absDiff = abs(self.getRoot().getHeight() - lst.getRoot().getHeight())
+		shorterTree = self
+		LongerTree = lst
+
+		b = LongerTree.getRoot()
+		while b.getHeight() > shorterTree.getRoot().getHeight():
+			b = b.getLeft()
 		
-		LastNodeInSelf = self.retrieveNode(self.size)
-		firstNodeInLst = lst.retrieveNode(self.size)
+		maxInShort = shorterTree.retrieveNode(shorterTree.size - 1)
+		maxInShort.getParent().right = AVLNode.virtualNode()
+
+		# if be is the root of lst
```

Comment:
comments are flipped, you write "if b is the root" but the condition is true iff b is not the root.